src/main/java/org/jabref/logic/integrity/IntegrityCheck.java

```diff
@@ -32,9 +45,36 @@ public IntegrityCheck(BibDatabaseContext bibDatabaseContext,
         this.journalAbbreviationRepository = Objects.requireNonNull(journalAbbreviationRepository);
         this.enforceLegalKey = enforceLegalKey;
         this.allowIntegerEdition = allowIntegerEdition;
+        initCheckers(bibDatabaseContext, bibtexKeyPatternPreferences, journalAbbreviationRepository);
     }
 
-    public List<IntegrityMessage> checkDatabase() {
+    private void initCheckers(BibDatabaseContext bibDatabaseContext, BibtexKeyPatternPreferences bibtexKeyPatternPreferences, JournalAbbreviationRepository journalAbbreviationRepository) {
```

Comment:
To be honest, I don't see any advantage of this refactoring. Just makes the code more complex and harder to maintain in my opinion.

If you really feel like the original code needs a refactoring, then you create a list of all checkers that need to be run a) always, b) bibtex c)biblatex (still I would create these lists only in the checkdatabase method). This would get ride of the repeated `check` methods below. But to be honest, I don't think this yields a more readable code either.

Reply:
We initialize the checkers once per database check - not once per entry check. In case of a 20k database, approx 100k less java objects in memory per quality check.

We ensured that each checker is stateless - thus, it can be reused for acroess entries.

I agree with your proposal though.

Reply:
With your proposal, the code is much more reable. 🎉 https://github.com/JabRef/jabref/pull/6333/commits/03dd2a0b59a63c17460a74cd12aa205edc628aad